test/integration/apiserver/apply/reset_fields_test.go

```diff
@@ -151,6 +151,7 @@ var resetFieldsSpecData = map[schema.GroupVersionResource]string{
 	gvr("resource.k8s.io", "v1alpha1", "resourceclaims"):                           `{"spec": {"resourceClassName": "class2name"}}`, // ResourceClassName is immutable, but that doesn't matter for the test.
 	gvr("resource.k8s.io", "v1alpha1", "resourceclaimtemplates"):                   `{"spec": {"spec": {"resourceClassName": "class2name"}}}`,
 	gvr("internal.apiserver.k8s.io", "v1alpha1", "storageversions"):                `{}`,
+	gvr("admissionregistration.k8s.io", "v1alpha1", "validatingadmissionpolicies"): `{"spec": {"paramKind": {"apiVersion": "apps/v1", "kind": "Deployment"}}}`,
```

Comment:
NOTE FOR REVIEWER: This test is to ensure `/status` endpoint prevent changes to `spec`. Here we attempt to change `spec`, the test will then ensure the change to `spec`  is reset and thus not taking effect.